### DISCUSSION TITLE
WP behind reverse proxy / public url

### DISCUSSION THREAD

**Discussion by Neolomoxem** [docker-library/wordpress#913] [General]
Hello everyone, quick basic question:
Im running WP via docker compose on a public url. This involves a NGINX proxy. When accessing my site, I noticed that most request fail, due to WP generating URLs for localhost:8080. After changing WP_HOME and WP_SITEURL in wp-config.php (inside the container) things work fine. I'm planning to persist the change by mounting a custom wp-content.php into the container.

Is this the best way to go about it, or am I missing something?

**Comment by tianon** [comment]
The cleanest way is to set up your reverse proxy to send the `X-Forwarded-For` (and `X-Forwarded-Proto`) header so that WordPress knows it's being proxied, but with the caveat that if you've already set up the instance, WordPress has a habit of embedding the initial setup URL in the database in various places (and then subsequently in pages, posts, etc when using the WYSIWYG editor).
